LibertyDSNP/example-client
# Features/create identity registration #178933068

Purpose
---------------
Functionality only, no styling, creates a registration on first time logging into DSNP network via example client. 
[Part of #178933068](https://www.pivotaltracker.com/story/show/178933068)


Solution
---------------
### In:
The first time logging into the example client, asks you to create a new handle and then finishes up the login process.

### Out:
* Styling to match design
* Reconciliation with choice of handles on first login if registration(s) already exist
* Checking if handle exists. This will be in a separate PR.

with @acruikshank , we will follow up in his PR by merging components and doing styling.

Change summary
---------------
- [x] New "Register" component that has a form for creating a new handle, and calls sdk.createRegistration with choice of handle and returns new DSNP userId to the Login component so it can finish up login and update user and profile state.
- [x] Basic UI w/o real styling 
- [x] if registration fails, unset everything so user can start over 

Steps to Verify
----------------
1.  Do setup of local hardhat node + static server + npm run start on this branch
1.  Login with a new account through Metamask that has no existing registration
1.  See the create handle form popup, enter a handle and submit
1.  Verify that the button reads Log out in the top right
1.  Verify that your new social ID, new handle and icon appear
1.  If you've run the populate script, verify that you can see posts in All Posts.

Additional details / screenshot
----------------
- Any supplemental pictures or material
![Screen Shot 2021-08-18 at 11 44 47 AM](https://user-images.githubusercontent.com/502640/129964918-9514d54f-6deb-4d92-aeb3-9a1bbc25c06b.png)
![giphy1](https://user-images.githubusercontent.com/502640/129989420-bb6cc7a6-0bbb-483e-8c58-711cd44192a2.gif)
